util.c

```diff
@@ -39,38 +48,88 @@ void safe_buffer_free(struct safe_buffer * sb)
 	}
 }
 
-size_t safe_buffer_pad(struct safe_buffer * sb, size_t count)
-{
-	size_t rc = 0;
-	unsigned char zero = 0;
+void safe_buffer_check_used_size(size_t proposed_size) {
+	if (proposed_size > 64*1024) {
+		flog(LOG_ERR, "Requested buffer too large for any possible IPv6 ND, even with jumbogram.  Exiting.");
+		exit(1);
+	}
+}
 
-	while (count--) {
-		rc += safe_buffer_append(sb, &zero, 1);
+void safe_buffer_resize(struct safe_buffer * sb, size_t new_capacity)
+{
+	// This may slightly exceed 64KiB at some points.
+	if (sb->allocated < new_capacity) {
+		sb->allocated = new_capacity;
+		sb->buffer = realloc(sb->buffer, sb->allocated);
 	}
+}
+
+void safe_buffer_expand(struct safe_buffer * sb, size_t additional_capacity)
+{
+	safe_buffer_resize(sb, sb->allocated + additional_capacity);
+}
 
-	return rc;
+size_t safe_buffer_pad(struct safe_buffer * sb, size_t count)
+{
+	safe_buffer_check_used_size(sb->used + count);
```

Comment:
Seems like safe_buffer_check_used_size should be called in safe_buffer_resize instead of being called separately.  Does it make sense to not check the size?  If not, then have the resize function always call this.